src/Assignment1/AssignmentService.java

```diff
@@ -0,0 +1,164 @@
+package Assignment1;
+
+public class AssignmentService {
+
+  private ItemModel model;
+
+  public AssignmentService() {}
+
+  public AssignmentService(ItemModel model) {this.model = model;}
+
+  // calculate tax methods
+  // method to calculate raw tax.
+  double rawTax(double total) {
+    if (total < 0) throw new RuntimeException("Total cant be -ve");
+    double tax = 0;
+    tax += total * 12.50 / 100;
+    return tax;
+  }
+
+  // method to calculate manufactured tax.
+  double manufacturedTax(double total) {
+    if (total < 0) throw new RuntimeException("Total cant be -ve");
+    double tax = 0;
+    tax += total * 12.5 / 100;
+    tax += (total + tax) * 2.0 / 100;
+    return tax;
+  }
+
+  // method to calculate imported tax.
+  double importedTax(double total) {
+    if (total < 0) throw new RuntimeException("Total cant be -ve");
+    double tax = total * 0.1;
+    if (total <= 100) tax += 5;
+    else if (total <= 200) tax += 10;
+    else tax += (total + tax) * 0.05;
+    return tax;
+  }
+
+  // method to calculate total cost.
+  double calculateCost() {
+    double totalCost = model.getPrice() * model.getQuantity();
+    double tax = 0;
+    if (model.getType().equals(ItemModel.types.RAW)) tax += rawTax(totalCost);
+    if (model.getType().equals(ItemModel.types.MANUFACTURED))
+      tax += manufacturedTax(totalCost);
+    if (model.getType().equals(ItemModel.types.IMPORTED)) tax += importedTax(totalCost);
+    model.tax = tax;
+    return totalCost+tax;
+  }
+
+  String extractWord(String inputString, int pos) {
+    if (inputString.isEmpty() || pos < 0 || pos >= inputString.length())
+      throw new RuntimeException("Invalid input");
+    StringBuilder name = new StringBuilder();
+    while (pos < inputString.length() && inputString.charAt(pos) != '-') {
+      name.append(inputString.charAt(pos++));
+    }
+    return name.toString();
+  }
+
+  // validate input method
+  boolean isInputValid(String inputString) {
```

Comment:
move all validations into separate utility class so that multiple services can reuse the validation code